### ISSUE TITLE
linter: `no-else-return` false positive on wrong branch

### ISSUE THREAD

**Issue by camchenry** [oxc-project/oxc#6364]
### What version of Oxlint are you using?

0.9.10

### What command did you run?

bunx oxlint@latest -D all -A no-unused-vars -A sort-keys -A no-null -A no-undefined -A explicit-function-return-type -A no-magic-numbers -A no-ternary -A no-explicit-any -A no-const-enum -A prefer-enum-initializers -A no-optional-chaining --ignore-pattern compliance

### What does your `.oxlint.json` config file look like?

N/A

### What happened?

While doing some linter testing on the https://github.com/angular/angular repository, I noticed that the `no-else-return` lint rule was correctly reporting an issue, but was reporting it for a line that is actually an `else-if` block.

Specifically, this code:
```typescript
export const getPropType = (prop: unknown): PropType => {
  if (isSignal(prop)) {
    prop = prop();
  }

  if (prop === undefined) {
    return PropType.Undefined;
  }
  if (prop === null) {
    return PropType.Null;
  }
  if (prop instanceof HTMLElement) {
    return PropType.HTMLNode;
  }
  const type = typeof prop;
  if (type in commonTypes) {
    return commonTypes[type as keyof typeof commonTypes];
  }
  if (type === 'object') {
    if (Array.isArray(prop)) {
      return PropType.Array;
    } else if (prop instanceof Set) {
      return PropType.Set;
    } else if (prop instanceof Map) {
      return PropType.Map;
    } else if (Object.prototype.toString.call(prop) === '[object Date]') {
      return PropType.Date;
    } else if (prop instanceof Node) {
      return PropType.HTMLNode;
    } else {
      return PropType.Object;
    }
  }
  return PropType.Unknown;
};
```

is reported as:

```
× eslint(no-else-return): Unnecessary 'else' after 'return'.
    ╭─[devtools/projects/ng-devtools-backend/src/lib/state-serializer/prop-type.ts:48:7]
 47 │     if (Array.isArray(prop)) {
 48 │       return PropType.Array;
    ·       ───────────┬──────────
    ·                  ╰── This consequent block always returns,
 49 │     } else if (prop instanceof Set) {
    ·      ───┬──
    ·         ╰── Making this `else` block unnecessary.
 50 │       return PropType.Set;
    ╰────
  help: Remove the `else` block, moving its contents outside of the `if` statement.
```

I believe this is incorrect, because the following else is part of an `if` statement, so it doesn't matter if the prior `if` always returns. It should actually be reporting this code as the error, since all previous `if` statements return:

```
} else {
  return PropType.Object;
}
```

**Comment by camchenry** [comment]
Ah actually I didn't read the configuration on this rule closely enough, it looks like it already allows for `allowElseIf: true` which would fix this. Looks to be working as intended.
